source/game/kart/KartCollide.cc

```diff
@@ -282,4 +394,54 @@ void KartCollide::applySomeFloorMoment(f32 down, f32 rate, CollisionGroup *hitbo
     }
 }
 
+bool KartCollide::FUN_805B6A9C(CollisionData &collisionData, Hitbox &hitbox,
+        EGG::BoundBox3f &minMax, EGG::Vector3f &relPos, s32 &count, Field::KCLTypeMask &maskOut,
+        const Field::CourseColMgr::CollisionInfo &colInfo) {
+    if (!!(maskOut & KCL_TYPE_FLOOR)) {
+        collisionData.floorNrm += colInfo.floorNrm;
+        collisionData.floor = true;
+    }
+
+    EGG::Vector3f tangentOff = colInfo.tangentOff;
+    minMax.min = minMax.min.minimize(tangentOff);
+    minMax.max = minMax.max.maximize(tangentOff);
+    tangentOff.normalise();
+
+    relPos += hitbox.relPos();
+    relPos += -hitbox.radius() * tangentOff;
+    ++count;
+
+    return false;
+}
+
+void KartCollide::applyBodyCollision(CollisionData &collisionData, const EGG::Vector3f &movement,
+        const EGG::Vector3f &posRel, s32 count) {
+    setPos(pos() + movement);
+
+    if (!collisionData.floor) {
+        collisionData.movement = movement;
+    }
+
+    f32 rotFactor = 1.0f / static_cast<f32>(count);
```

Comment:
Do we accept a potential division by zero?

Reply:
No check for that in-game. And we can prove this is a safe operation.

- `applyBodyCollision` will only be called if `FUN_805B6A9C` returns false
- In the scenario where `FUN_805B6A9C` returns false, it is guaranteed that `count` is incremented at least once
- Count is initialized to 0 in `calcBodyCollision`
- QED